### ISSUE TITLE
xds_bootstrap_test fails under Bazel RBE Windows (both dbg and opt) 

### ISSUE THREAD

**Issue by markdroth** [grpc/grpc#20541]
https://source.cloud.google.com/results/invocations/ad826d9a-7e6b-43ac-bd17-3216b9f924f4/targets/%2F%2Ftest%2Fcore%2Fclient_channel:xds_bootstrap_test/tests

There are 3 test failures that happen fairly repeatably, but the most interesting one is XdsBootstrapTest.BadChannelCredsContents, which fails with the following error:

```
unknown file
SEH exception with code 0xc00000fd thrown in the test body.
```

I'll change the BUILD file to not run this test under Windows for now, but we should re-enable that once we figure out what's causing it.

**Comment by nicolasnoble** [comment]
I managed to reproduce it. There's an infinite recursion happening. Here's the stacktrace:

```
xds_bootstrap_test.exe!add_errs(grpc_error * err, char * * s, unsigned __int64 * sz, unsigned __int64 * cap) Line 698
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(698)
xds_bootstrap_test.exe!errs_string(grpc_error * err) Line 710
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(710)
xds_bootstrap_test.exe!grpc_error_string(grpc_error * err) Line 761
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(761)
xds_bootstrap_test.exe!add_errs(grpc_error * err, char * * s, unsigned __int64 * sz, unsigned __int64 * cap) Line 696
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(696)
xds_bootstrap_test.exe!errs_string(grpc_error * err) Line 710
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(710)
xds_bootstrap_test.exe!grpc_error_string(grpc_error * err) Line 761
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(761)
xds_bootstrap_test.exe!add_errs(grpc_error * err, char * * s, unsigned __int64 * sz, unsigned __int64 * cap) Line 696
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(696)
xds_bootstrap_test.exe!errs_string(grpc_error * err) Line 710
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(710)
xds_bootstrap_test.exe!grpc_error_string(grpc_error * err) Line 761
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(761)
xds_bootstrap_test.exe!add_errs(grpc_error * err, char * * s, unsigned __int64 * sz, unsigned __int64 * cap) Line 696
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(696)
xds_bootstrap_test.exe!errs_string(grpc_error * err) Line 710
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(710)
xds_bootstrap_test.exe!grpc_error_string(grpc_error * err) Line 761
	at c:\sources\grpc\src\core\lib\iomgr\error.cc(761)
xds_bootstrap_test.exe!grpc_core::testing::XdsBootstrapTest_BadChannelCredsContents_Test::TestBody() Line 257
	at c:\sources\grpc\test\core\client_channel\xds_bootstrap_test.cc(257)
xds_bootstrap_test.exe!testing::internal::HandleSehExceptionsInMethodIfSupported<testing::Test,void>(testing::Test * object, void(testing::Test::*)() method, const char * location) Line 2424
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2424)
xds_bootstrap_test.exe!testing::internal::HandleExceptionsInMethodIfSupported<testing::Test,void>(testing::Test * object, void(testing::Test::*)() method, const char * location) Line 2475
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2475)
xds_bootstrap_test.exe!testing::Test::Run() Line 2521
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2521)
xds_bootstrap_test.exe!testing::TestInfo::Run() Line 2693
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2693)
xds_bootstrap_test.exe!testing::TestSuite::Run() Line 2823
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2823)
xds_bootstrap_test.exe!testing::internal::UnitTestImpl::RunAllTests() Line 5343
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(5343)
xds_bootstrap_test.exe!testing::internal::HandleSehExceptionsInMethodIfSupported<testing::internal::UnitTestImpl,bool>(testing::internal::UnitTestImpl * object, bool(testing::internal::UnitTestImpl::*)() method, const char * location) Line 2424
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2424)
xds_bootstrap_test.exe!testing::internal::HandleExceptionsInMethodIfSupported<testing::internal::UnitTestImpl,bool>(testing::internal::UnitTestImpl * object, bool(testing::internal::UnitTestImpl::*)() method, const char * location) Line 2475
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(2475)
xds_bootstrap_test.exe!testing::UnitTest::Run() Line 4930
	at c:\sources\grpc\third_party\googletest\googletest\src\gtest.cc(4930)
xds_bootstrap_test.exe!RUN_ALL_TESTS() Line 2473
	at c:\sources\grpc\third_party\googletest\googletest\include\gtest\gtest.h(2473)
xds_bootstrap_test.exe!main(int argc, char * * argv) Line 335
	at c:\sources\grpc\test\core\client_channel\xds_bootstrap_test.cc(335)
```

**Comment by nicolasnoble** [comment]
Narrowing it down a bit, it looks like the culprit is somewhere around that code:

```C
  grpc_error* error = GRPC_ERROR_NONE;
  grpc_core::XdsBootstrap bootstrap(slice, &error);
  gpr_log(GPR_ERROR, "%s", grpc_error_string(error));
  ASSERT_TRUE(error != GRPC_ERROR_NONE);
  std::regex e(
      std::string("errors parsing \"xds_server\" object(.*)"
                  "errors parsing \"channel_creds\" object(.*)"
                  "\"type\" field is not a string(.*)"
                  "duplicate \"type\" field(.*)"
                  "\"config\" field is not an object(.*)"
                  "duplicate \"config\" field"));
  VerifyRegexMatch(error, e);
```

**Comment by markdroth** [comment]
I'm pretty sure this is fixed in #20606.
